### PR TITLE
Moving topic editing into a dialog to improve topics display 

### PR DESCRIPTION
**Pull request by kathy-t**
**Description**
This PR improves the topics display on the private entry page by only displaying the selected topic with a bubble indicating the type of topic and moving the topic selection into a new dialog where users can see all 3 types of topics, edit the manual topic, and select the topic of choice. By moving it into a new dialog, we have more space to explain each type of topic without relying on tooltips. The main Info panel is also easier to read since there's no longer 4 lines of topic-related info.

**Before**
Screenshot 2024-07-05 at 11-41-57 Dockstore My Workflows

**After**
GitHub App workflow - notice how the manual topic is not editable and the caption says to edit it in the .dockstore.yml
Screenshot 2024-07-05 at 11-32-41 Dockstore Workflow

Screenshot 2024-07-05 at 11-32-58 Dockstore Workflow

Hosted workflow - notice how the manual topic is editable and that hosted entries only have two options: manual and AI.
Screenshot 2024-07-05 at 11-33-41 Dockstore Workflow

Screenshot 2024-07-05 at 11-33-51 Dockstore Workflow

Full workflow (traditional registration)
Screenshot 2024-07-05 at 11-35-08 Dockstore Workflow

**Review Instructions**
Verify that you can select different types of topics. On the private entry page, you should see the selected topic and a bubble indicating the type of topic. On the public page, you should only see the topic (with no bubble unless it's an AI topic). Repeat this for a variety of entry types (GitHub app workflow/tool, hosted workflow/tool, legacy workflow/tool).

**Issue**
https://github.com/dockstore/dockstore/issues/5727
https://ucsc-cgl.atlassian.net/browse/DOCK-2483

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

### CODE REVIEW

File: src/app/shared/entry/info-tab-topic/display-topic/display-topic.component.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  @Input() entry: DockstoreTool | Workflow;
+  @Input() disableEditing: boolean;
+  constructor(private sessionQuery: SessionQuery, private dialog: MatLegacyDialog) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.isPublic$ = this.sessionQuery.isPublic$;
+    this.entryTypeMetadata = this.entry.entryTypeMetadata;
+    this.isGitHubAppEntry = (this.entry as Workflow | DockstoreTool).mode === Workflow.ModeEnum.DOCKSTOREYML; // The workflow and tool enums are confirmed to be equal in a test.
```
Comment (coverbeck) [inline src/app/shared/entry/info-tab-topic/display-topic/display-topic.component.ts:38]: Do you mean you manually tested it, or there's an actual test (I couldn't find a test in this PR)?

If there's not an actual test, I'd suggest adding one. Probably being paranoid and it's extremely unlikely, but maybe someday, something changes in OpenAPI code generation, or we rename one enum and overlook renaming the other -- the compiler wouldn't catch either of those.

Or be verbose and compare against both enums, and/or use a type guard.

File: src/app/shared/entry/info-tab-topic/edit-topic/edit-topic-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+  constructor(
+    public dialogRef: MatLegacyDialogRef<EditTopicDialogComponent>,
+    public editTopicDialogService: EditTopicDialogService,
+    @Inject(MAT_LEGACY_DIALOG_DATA) public data: EditTopicDialogData
+  ) {
+    this.entry = data.entry;
+    this.entryType = data.entry.entryType;
+    this.entryTypeMetadata = data.entry.entryTypeMetadata;
+    this.isGitHubAppEntry = (data.entry as Workflow | DockstoreTool).mode === Workflow.ModeEnum.DOCKSTOREYML; // The workflow and tool enums are confirmed to be equal in a test.
```
Comment (coverbeck) [inline src/app/shared/entry/info-tab-topic/edit-topic/edit-topic-dialog.component.ts:80]: Same as other comment

**Review by coverbeck — APPROVED** [review]

**Comment by denis-yuen** [comment]
Before reviewing code, I like the screenshots except for the "edit" button. It's a bit understated given that it leads to a dialogue box. For example, the actions button is a bit more inviting(?)  
Screenshot from 2024-07-05 15-04-24

**Comment by kathy-t** [comment]
> Before reviewing code, I like the screenshots except for the "edit" button. It's a bit understated given that it leads to a dialogue box. For example, the actions button is a bit more inviting(?)

Hmm, I see what you're saying, but the edit button opening a dialog is an existing behaviour on other pages like the organizations page

image

Unless we prefer a different wording for it like "Configure" with a gear icon?

**Comment by denis-yuen** [comment]
> Hmm, I see what you're saying, but the edit button opening a dialog is an existing behaviour on other pages like the organizations page
> Unless we prefer a different wording for it like "Configure" with a gear icon?

Fair point. Or maybe something like the old preview badge
https://github.com/dockstore/dockstore-ui2/pull/1012

**Review by svonworl — APPROVED** [review]
Looks good!

**Comment by kathy-t** [comment]
> Fair point. Or maybe something like the old preview badge
https://github.com/dockstore/dockstore-ui2/pull/1012

@denis-yuen I believe we use "preview" as a way to indicate that a feature is still under development (ex: services and previously notebooks) so I'm hesitant to use that term in another context.

I think I will keep it as Edit for now to match the existing behaviour on other pages unless you and others feel strongly about changing it

**Comment by denis-yuen** [comment]
> I think I will keep it as Edit for now to match the existing behaviour on other pages unless you and others feel strongly about changing it

I think that's fine, but do create a ticket to explore something like a "new!" badge or something. 
This all seems a little hidden and I think we'd like to show-off this work when released

**Review by denis-yuen — APPROVED** [review]

**Comment by kathy-t** [comment]
> I think that's fine, but do create a ticket to explore something like a "new!" badge or something. This all seems a little hidden and I think we'd like to show-off this work when released

Created https://ucsc-cgl.atlassian.net/browse/SEAB-6561. This may be nice to have for other new features in the UI as well

**Pull request merged**